### ISSUE TITLE
As a user and a developer, I want to display Japanese characters in charts

### ISSUE THREAD

**Issue by kyokonishito** [pixiedust/pixiedust#760]
As a user and a developer, I want to display Japanese characters in charts

### Expected behavior
* We can see all Japanese characters in labels and legends of all charts correctly.

### Actual behavior

* All Japanese characters in labels and legends of all charts are displayed with squares.

### Steps to reproduce the behavior
import pixiedust
import pandas as pd
df = pd.DataFrame([["四月", 26],["May", 10],["June", 5]],  columns=['key', 'value'])
display(df)

**Comment by vabarbosa** [comment]
@kyokonishito can you confirm this is specifically when using the `matplotlib` renderer. if you switch to `bokeh` do you properly see the characters?

**Comment by vabarbosa** [comment]
@DTAIEB i believe the issue is with the default font used by matplotlib does not support Japanese, Chinese, etc characters:

- https://matplotlib.org/users/dflt_style_changes.html#fonts

we  could:

1. start adding/including fonts packages. 
1. implement an option like: `display(df, font_name='someFontName')` and just pass the `font_name` to [`matplotlib` to use](https://jdhao.github.io/2017/05/13/guide-on-how-to-use-chinese-with-matplotlib/#use-a-font)
1. make no changes and just recommend users choose a different renderer when dealing with international characters

my vote would be for **2** but this would leave it up to the user to make sure they have the preferred font already installed in their system and know the name so it can be passed in `display`.

**Comment by kyokonishito** [comment]
@vabarbosa I confirmed `bokeh` can be displayed Japanese. However the pie chart does not have `bokeh` option.
I would like to have your option 2. We can get OSS fonts can be displayed Japanese. When I display wordcloud in Japanese on Watson Studio, I run following code. I set the font path to wordcloud:
```
import os
if not os.path.exists('/home/dsxuser/work/ipaexg00301/ipaexg.ttf'):
    !wget https://oscdl.ipa.go.jp/IPAexfont/ipaexg00301.zip
    !unzip ipaexg00301.zip
else:
    print('IPA font already installed')

from wordcloud import WordCloud

wordcloud = WordCloud(background_color="white", 
font_path="/home/dsxuser/work/ipaexg00301/ipaexg.ttf").generate(text)

```

**Comment by vabarbosa** [comment]
@kyokonishito as i think about this more i am wondering if it is best to keep with pixiedust convention and put the `font_path` field in the Chart Options dialog? since it is only specific to matplotlib we can show the field in options dialog when using matplotlib and hide it when not using matplotlib.

with either case, i think we should probably also store the `font_path` value in the pixiedust user preference db, so once you set it you do not need to keep setting it for every chart.

i want to wait for feedback from @DTAIEB before committing any changes. however, in the meantime you can use try using my working branch which has a temporary fix to keep you from being blocked:

1. install pixiedust from my working branch:
```
!pip install --upgrade --no-deps git+https://github.com/pixiedust/pixiedust.git@va-working-branch#egg=pixiedust
```
2. restart kernel
3. add `font_path` parameter to `display`
```
display(df, font_path='/Users/va/blue/tools/jp-font/ipaexg00301/ipaexg.ttf')
```
and you should see characters displayed properly:

![image](https://user-images.githubusercontent.com/13156555/55575712-90496100-56dd-11e9-96e1-2a3f7e5402e1.png)

**Comment by kyokonishito** [comment]
@vabarbosa Thank you! I tried your working branch. It works fine. 
However there is one strange issue. I had an error if I set the fon_tpath with a String variable.
```
import pixiedust
import pandas as pd

df = pd.DataFrame([["四月", 26],["May", 10],["June", 5]],  columns=['key', 'value'])
#display(df, font_path='/home/dsxuser/work/ipaexg00301/ipaexg.ttf') #this works fine !

jp_font_path='/home/dsxuser/work/ipaexg00301/ipaexg.ttf'
display(df, font_path=jp_font_path) # this has [Errno 2] No such file or directory: 'jp_font_path'
```
<img width="909" alt="pixiedust2019-04-05_11-02-38" src="https://user-images.githubusercontent.com/40123488/55599500-170c4580-5793-11e9-9f67-4b7458944906.png">

**Comment by DTAIEB** [comment]
> @DTAIEB i believe the issue is with the default font used by matplotlib does not support Japanese, Chinese, etc characters:
> 
> * https://matplotlib.org/users/dflt_style_changes.html#fonts
> 
> we could:
> 
> 1. start adding/including fonts packages.
> 2. implement an option like: `display(df, font_name='someFontName')` and just pass the `font_name` to [`matplotlib` to use](https://jdhao.github.io/2017/05/13/guide-on-how-to-use-chinese-with-matplotlib/#use-a-font)
> 3. make no changes and just recommend users choose a different renderer when dealing with international characters
> 
> my vote would be for **2** but this would leave it up to the user to make sure they have the preferred font already installed in their system and know the name so it can be passed in `display`.

@vabarbosa Yes I agree with #2. To make it easier, perhaps we could also have a drop down contextual option to let users pick from some of the most common fonts.

**Comment by kyokonishito** [comment]
@vabarbosa now,  the  error about the font_path with a String variable is not occured. I'm not sure the cause, but it works fine now.
I will introduce PixieDust in my session on a Japan IBM Cloud Community event on Apr 26. May I introduce your  temporary fix for Japanese attendees in my session if the module is not fixed by Apr 26?

**Comment by vabarbosa** [comment]
@kyokonishito i will keep posted, but yes feel free to use the temporary fix in my branch in the meantime.
